gen/definitions/deploy_template.yaml

```diff
@@ -0,0 +1,98 @@
+---
+name: Deploy Template
+rest_endpoint: /dna/intent/api/v2/template-programmer/template/deploy
+no_delete: true
+no_update: true
+no_read: true
+doc_category: Templates
+test_tags: [TEMPLATES]
+attributes:
+  - model_name: templateId
+    type: String
+    id: true
+    description: Main template UUID of versioned template
+    test_value: catalystcenter_template_version.example.id
+    example: 12345678-1234-1234-1234-123456789012
+  - model_name: forcePushTemplate
+    type: Bool
+    description: Force Push Template
+    example: false
+  - model_name: isComposite
+    type: Bool
+    description: Composite template flag
+    example: false
+  - model_name: mainTemplateId
+    type: String
+    description: Main template UUID of versioned template
```

Comment:
What is the difference between this and `templateId`? Anything we can put in the description to explain this?

Reply:
I think this is being use when you are using composite templates: 
mainTemplateId = Composite Template ID (Also can be considered as a parent template ID)

```
Create a Composite Template
Two or more regular templates are grouped into a composite sequence template. You can create a composite sequential template for a set of templates, which are applied collectively to devices. For example, when you deploy a branch, you must specify the minimum configurations for the branch router. The templates that you create can be added to a single composite template, which aggregates all the individual templates that you need for the branch router. You must specify the order in which templates that are in the composite template are deployed to devices.
```

we can keep it simple and put description: Composite Template ID

Reply:
i added Composite Template ID as description